components/StyledInputTags.js

```diff
@@ -135,27 +136,18 @@ const StyledInputTags = ({ suggestedTags, value, onChange, renderUpdatedTags, de
   const scrollerRef = React.useRef();
   const [isOpen, setOpen] = React.useState(false);
   const [tags, setTags] = React.useState(getOptions(value || defaultValue));
-  useGlobalBlur(wrapperRef, outside => {
-    if (outside) {
-      handleClose();
-    }
-  });
 
-  const handleClose = () => {
-    onChange(tags);
-    setOpen(false);
-  };
-  const handleToggleInput = () => {
-    isOpen ? handleClose() : setOpen(true);
-  };
-  const addTag = tag => setTags(uniqBy([...tags, { label: tag.toLowerCase(), value: tag.toLowerCase() }], 'value'));
-  const removeTag = (tag, update) => {
-    const updatedTags = tags.filter(v => v.value !== tag);
-    setTags(updatedTags);
-    if (update) {
-      onChange(updatedTags);
+  const handleClose = React.useCallback(() => {
+    if (isOpen) {
```

Comment:
This is the core fix: this function is called by `useGlobalBlur` on each click, but if `isOpen` is false there's no reason to trigger this code.